lib/encoding/yaml/parser.toit

```diff
@@ -0,0 +1,1689 @@
+    return null
+
+  s-double-break n/int -> List?:
+    if rune := s-double-esscaped n: return rune
+    if folded := s-flow-folded n:
+      runes := List
+      folded.do --runes: runes.add it
+      return runes
+    return null
+
+  s-double-esscaped n/int -> List?:
+    try-parse:
+      white-spaces := repeat: s-white
+      if match-char C-ESCAPE_ and
+         b-non-content and
+         (repeat: l-empty n FLOW-IN_) and
+         s-flow-line-prefix n:
+        return white-spaces
+    return null
+
+  ns-single-char -> int?:
+    try-parse:
+      if not s-white:
+        if rune := nb-single-char:
+          return rune
+    return null
+
+  nb-single-char -> int?:
+    try-parse:
+      if rune := c-quoted-quote: return '\''
+      if rune := nb-json:
+        if rune != C-SINGLE-QUOTE_: return rune
+    return null
+
+  ns-double-char -> int?:
+    try-parse:
+      if not s-white:
+        if rune := nb-double-char:
+          return rune
+    return null
+
+  nb-double-char -> int?:
+    if rune := c-ns-esc-char: return rune
+    try-parse:
+      if rune := nb-json:
+        if rune != C-ESCAPE_ and rune != C-DOUBLE-QUOTE_: return rune
+    return null
+
+  static SIMPLE-ESCAPES ::= { '0': 0, 'a': '\a', 'b': '\b', 't': '\t', '\t': '\t', 'n': '\n',
+                              'v': '\v', 'f': '\f', 'r': '\r', 'e': 0x18, ' ': ' ', '"': '"',
+                              '/': '/', '\\': '\\', 'N': 0x85, '_': 0x0a, 'L': 0x2028,
+                              'P': 0x2029 }
+  static SIMPLE-ESCAPE-KEYS ::= keys-as-set_ SIMPLE-ESCAPES
+
+  c-ns-esc-char -> int?:
+    try-parse:
+      if match-char C-ESCAPE_:
+        if rune := match-chars SIMPLE-ESCAPE-KEYS:
+          return SIMPLE-ESCAPES[rune]
+        if match-char 'x': if res := match-hex 2: return res
+        if match-char 'u':
+          if res := match-hex 4:
+            if 0xd800 <= res <= 0xdbff:
+              if part-2 := c-ns-esc-char:
+                 if not 0xdc00 <= part-2 <= 0xdfff:
+                  set-error "INVALID_SURROGATE_PAIR"
+                  return null
+                 return 0x10000 + ((res & 0x3ff) << 10) | (part-2 & 0x3ff)
+            else:
+              return res
+        if match-char 'U': if res := match-hex 8: return res
+    return null
+
+  // Space
+  b-as-space: return b-break
+
+  b-as-line-feed: return b-break
+
+  l-empty n c:
+    try-parse:
+      if (s-line-prefix n c or s-indent-less-than n) and b-as-line-feed:
+        return true
+    return false
+
+  s-line-prefix n c:
+    if c == BLOCK-OUT_ or c == BLOCK-IN_: return s-block-line-prefix n
+    else: return s-flow-line-prefix n
+
+  s-separate n c:
+    if c == BLOCK-KEY_ or c == FLOW-KEY_: return s-separate-in-line
+    return s-separate-lines n
+
+  s-separate-lines n:
+    try-parse:
+      if s-l-comments and s-flow-line-prefix n: return true
+    return s-separate-in-line
+
+  s-flow-line-prefix n:
+    return s-indent n and (optional: s-separate-in-line)
```

Comment:
I don't think this is correct. It would return false if `s-separate-in-line` is false. However that's only optional.
```suggestion
    if s-indent n:
      optional: s-separate-in-line
      return true
    return null
```
(I'm returning `true`, but I'm not sure that's the correct return value.

Reply:
optional will return EMPTY_NODE_ (aka a truthful value) when not called with --or-null. So it is correct.